### ISSUE TITLE
请问 batchnorm var 为 0 的时候是怎么处理的？

### ISSUE THREAD

**Issue by ChunyanZhao** [Tencent/ncnn#302]
我在 batchnorm 的参数为 0 的时候，得到的输出是 nan，是不是需要做特殊设置？

**Comment by nihui** [comment]
mxnet 模型转换过来的？

**Comment by ChunyanZhao** [comment]
从 caffe 转过来的， mean 和 var 全 0， scale 是 1 .  检查的时候发现是转换的时候出错， 在 caffe2ncnn.cpp  441行:
      `float scale_factor = 1 / binlayer.blobs(2).data().data()[0];`
我转换的时候，读到的 binlayer.blobs(2).data().data()[0] 是 0.

**Comment by Josuercuevas** [comment]
If you check line 100 in file batch_norm_layer.cpp at the **original caffe repo**: [BatchNorm layer](https://github.com/BVLC/caffe/blob/master/src/caffe/layers/batch_norm_layer.cpp), this problem when binlayer.blobs(2).data().data()[0] = 0 should be handled as follows:
```c
float scale_factor = binlayer.blobs(2).data().data()[0] == 0 ? 0 : 1 / binlayer.blobs(2).data().data()[0];
```
